docs/code-standards.md

```diff
@@ -0,0 +1,35 @@
+# Code & Documentation Standards
+
+This document details our goals for code quality, coverage and documentation standards. At the time
+of this writing, these are standards that all of our code may not meet. We should always try to 
+improve our code to meet them.
+
+## Code Quality And Documentation Standards
+
+We aim to write clear code that is easy to read and comprehend. In particular, that goal means the
+following:
+
+1. All Go code must pass the Go linter checks
+    - We have these checks integrated into our CI system
+2. All Go code should pass the `go vet`
+    - These checks are not integrated into our CI system
```

Comment:
While I wouldn't suggest removing (1) and (2) above, I agree with what @MHBauer mentions [here](https://github.com/kubernetes-incubator/service-catalog/pull/696/files#r111459838). _All_ such checks are bundled together in `make verify`, so if for no reason other than convenience, we should mention that.